jwst/datamodels/tests/test_filetype.py

```diff
@@ -0,0 +1,18 @@
+import pytest
+
+from ...pipeline import Spec2Pipeline
+
+
+@pytest.fixture(scope='module')
+def fake_pipeline():
+    return Spec2Pipeline()
+
+
+def test_pipeline_raises_filenotfound(fake_pipeline, capsys):
+    with pytest.raises(RuntimeError):
+        fake_pipeline.run('file_does_not_exist.fits')
+
+    assert (
+            "FileNotFoundError: [Errno 2] No such file or directory: 'file_does_not_exist.fits'"
```

Comment:
Why not just check for "FileNotFoundError"?

Reply:
I think the `Spec2Pipeline` catches `FileNotFoundError` and raises a `RuntimeError`.  Not great.  That needs to be fixed.

Perhaps simpler test that doesn't mix up `jwst.datamodels` and `jwst.pipeline` in the tests would be to just to open a non-existent file using `datamodels.open` and one should get a nice `pytest.raises(FileNotFoundError)`?

Reply:
Turns out that error is captured and printed for the pipeline that I used (not sure if this is the case for all pipelines). For `Spec2Pipeline` if there are any errors during the course of executing the pipeline, the error is printed and then at the end a `RuntimeError` is raised. No other way to access it other than looking for it in the `stdout`

Reply:
Also the reason that I used that pipeline in the test is because in the issue, JP-1320, that was where the clarity was desired.

Reply:
@jdavies-st That's a good idea and preferable for sure. I wasn't sure where `check` was used, so I just defaulted to using a pipeline. I'll switch it out

Reply:
I just meant that you don't need to search for the entire string, that "FileNotFoundError" should be sufficient.  Searching on the entire string seems fragile to me.  For example, what if the error number differs on different operating systems?  What if in some version of Python they decide to use double quotes instead of single quotes around the file name?  What if they add a period after "Errno 2]"?  What if they decide to change from "No such file or directory" to "File <name> does not exist"?

Reply:
@philhodge Yeah I see you're point, but at the same time I think a similar argument could be made in that looking for just `FileNotFoundError` is non-specific. What if there's more than one of those for some reason?

Reply:
Either way, I think that issue will be solved by using `datamodels.open`

Reply:
Yeah, `Spec2Pipeline` captures all exceptions and then re-emits a `RuntimeError`.  This is bad needs fixing.

https://github.com/spacetelescope/jwst/blob/05ab872e7c833e08b9b3daa3f158f605bc71a3d6/jwst/pipeline/calwebb_spec2.py#L91-L112